torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -792,11 +792,15 @@ void LoopNest::inlineIntermediateBufs(bool allow_duplicated_work) {
       // tensors, always inline, bc we are not duplicating any work
       // and avoiding an intermediary buffer
       if (stores.size() == 1) {
-        auto store = dynamic_cast<Store*>(stores[0].s);
-        auto input_as_load = dynamic_cast<const Load*>(store->value());
-        if (input_as_load && input_bufs.count(input_as_load->buf())) {
-          bufs_to_inline.insert(buf);
-          continue;
+        if (auto store = dynamic_cast<Store*>(stores[0].s)) {
```

Comment:
How is this code related to the rest, is it for lowering with external calls?

Reply:
It's a fix for a bug in inliner related to ExternalCalls handling. I've added a test for this!